neuro_san/http_sidecar/handlers/health_check_handler.py

```diff
@@ -43,18 +43,17 @@ def initialize(self, forwarded_request_metadata: List[str]):
             self.set_header("Access-Control-Allow-Methods", "GET, OPTIONS")
             self.set_header("Access-Control-Allow-Headers", "Content-Type, Transfer-Encoding")
 
-        self.versions: Dict[str, Any] = self.determine_versions()
-
     async def get(self):
         """
         Implementation of GET request handler for API health check.
         """
 
         try:
+            versions: Dict[str, Any] = self.determine_versions()
             result_dict: Dict[str, Any] = {
                 "service": "neuro-san agents",
                 "status": "healthy",
-                "versions": self.versions
+                "versions": versions
             }
```

Comment:
Move logic for determining current versions under general try-catch block,
so we don't have some unhandled exception (not really likely, but) just flying out.
Perf impact: should be none, since "initialize" is called for every new incoming request anyway.